### ISSUE TITLE
Incrementation Distorts Percussion in ReadNoiseSample

### ISSUE THREAD

**Issue by TempoQuill** [pret/pokecrystal#924]
This issue ranges from SW97 all the way to Crystal.  In `ReadNoiseSample`, the first byte is split into a purely cosmetic hi nybble to make importing RBY noise data a matter of copy-paste, and a lo nybble for the length.

All the RBY percussion is already there, and drums that are two or more notes long indicate the same note length structure as in RBY.  However, because of the way `ReadNoiseSample` stores this first byte, this length is one frame longer than intended, meaning minimum note length is two instead of one.  This makes the old drums with two or more notes sound slower than they should be.

Would this be counted as a bug?

```	ld a, [de]
	inc de

	cp sound_ret_cmd
	jr z, .quit

	and $f
	inc a
	ld [wNoiseSampleDelay], a```

**Comment by aaaaaa123456789** [comment]
No, that's a conversion error. A length of zero is nonsense; the duration being forced to a range of 1-16 is reasonable and matches the behavior of the other channels.

**Comment by Rangi42** [comment]
If the GSC music sounds the way it's supposed to, then this isn't a bug.

**Comment by TempoQuill** [comment]
> No, that's a conversion error. A length of zero is nonsense;

The intended length is 1-16, but the `inc a` makes it so $x0 means two frames rather than one.

```	ld a, [wNoiseSampleDelay]
	and a
	jr z, ReadNoiseSample
	dec a
	ld [wNoiseSampleDelay], a```

When we arrive back at the very end of `HandleNoise`, `wNoiseSampleDelay` instructs the number of updates before we reach `ReadNoiseSample` again.  If `wNoiseSampleDelay` says zero, that means `ReadNoiseSample` is accessible at that moment.  If `wNoiseSampleDelay` is incremented beforehand, that means drum notes can never be shorter than two frames.  Instead of 1-16, it's 2-17.

**Comment by vulcandth** [comment]
Would changing this alter the way GSC music sounds? If so, then this becomes a subjective change.

I think in discord you mentioned that it is handled differently in RBY? What is the benefit to changing this? 

Also, **_if_** this is something we should document... it would probably be more of a design flaw imo.

**Comment by TempoQuill** [comment]
In RBY, drums were pretty much one of the only things that REALLY worked properly.  It had one byte for a length of 1-16 frames, designated by `2x`.  This was very consistent with the sound effect data, which was ALSO designated with `2x` on the first byte.

In GSC, Gen 1's percussion was imported as is, because of the cosmetic hi nybble.  By removing the `inc` instruction before storing the length in RAM, the Gen 1 sound effect would sound as they did in RBY.  In addition, the newer percussion becomes cleaner and clearer.

**Comment by Idain** [comment]
The "cleaner and clearer" is a subjective matter. Also, not being able t import Gen 1 songs into Gen 2 because of differences in drum sounds more like a design flaw than a bug, since they never planned for this to happen.

It _could_ be documented as a design flaw, but definitely not a bug.

**Comment by Idain** [comment]
It'd be cool if you posted the songs somewhere without the `inc a` instruction to see how they'd sound like so people could make a better judgement whether to "fix" it or not.

YouTube could be perfect for it, or even Soundcloud.

**Comment by vulcandth** [comment]
> YouTube could be perfect for it, or even Soundcloud.

Careful with Youtube. Good chance the original would get flagged for a copyright claim. Even the edited one, could get flagged.

**Comment by TempoQuill** [comment]
> I was under the impression the formats are quite different.

Different enough you couldn't copy-paste on HxD, but you could tie the two formats together somewhat with identical macros that work differently depending on the format.

**Comment by Idain** [comment]
All of this still sounds like a design flaw rather than a bug. After all, they never intented to use Gen 1 songs in GSC.

**Comment by Idain** [comment]
> Careful with Youtube. Good chance the original would get flagged for a copyright claim. Even the edited one, could get flagged.

@vulcandth, I was talking about only uploading the modified versions. Also, it's hard for Youtube to catch new accounts with almost no subs, so I wouldn't worry about it. There are 12+ years videos with Pokémon music and they haven't been flagged yet.

**Comment by vulcandth** [comment]
> @vulcandth, I was talking about only uploading the modified versions. Also, it's hard for Youtube to catch new accounts with almost no subs, so I wouldn't worry about it. There are 12+ years videos with Pokémon music and they haven't been flagged yet.

I don't know what his youtube status is. They could be Mr. Beast for all I know. Also, yes the risk is low.. but it is not zero. It it up to them to determine their own risk tolerance.

**Comment by Idain** [comment]
You have a good point there. A wiki tutorial for it with a disclaimer stating that it may affect Gen 2 songs as well.

**Comment by vulcandth** [comment]
@TempoQuill Based on the discussion that occurred in discord, I believe the determination from Rangi was that this may indeed be something that could be considered a bug/design flaw, and that the fix should be documented in a wiki page **first** so she may look it over further. It should be pretty easy to port from the wiki to the repo `docs/` later. Would you have time to start the wiki page?

**Comment by coltongit** [comment]
@Rangi42 would the wiki page be sufficient for this? Considering the PR's already been thanos'd I might as well ask
If it is, we can close this one

**Comment by Rangi42** [comment]
The wiki page was you, haven't heard from @TempoQuill re: what else should be added, cleaned up, whether it's accurate, etc.

**Comment by coltongit** [comment]
@TempoQuill Is this wiki page suitable for this now?
